src/user.ts

```diff
@@ -4,8 +4,18 @@ import { Table, tableFromIPC } from 'apache-arrow';
 export const isNode =
   typeof process !== 'undefined' && process.versions && process.versions.node;
 
-export abstract class BaseAtlasClass {
+export type LoadedObject<
+  T extends BaseAtlasClass<U>,
+  U extends Record<string, any>
+> = T & { i: U };
+
+export abstract class BaseAtlasClass<InfoType extends Record<string, any>> {
   user: AtlasUser;
+  // To avoid multiple calls, the first request sets the _info property.
+  protected _info: Promise<InfoType> | undefined;
+  // Once info resolves, it populates here.
+  protected _i: InfoType | undefined;
+
```

Comment:
I don't know if others feel this way but I don't love using info/_info and i/_i in these similar but subtly different ways. I think this will be very confusing to see in the code. Could we come up with two words here that better reflect the difference?

Reply:
maybe `populateInfo` and `getInfo`? Something along those lines?

Reply:
Incorporated